server/service/hosts.go

```diff
@@ -100,7 +113,38 @@ func listHostsEndpoint(ctx context.Context, request interface{}, svc fleet.Servi
 
 		hostResponses[i] = *h
 	}
-	return listHostsResponse{Hosts: hostResponses, Software: software}, nil
+	return listHostsResponse{Hosts: hostResponses, Software: software, MDMSolution: mdmSolution}, nil
+}
+
+func (svc *Service) AggregatedMDMSolutions(ctx context.Context, teamID *uint, mdmID uint) (*fleet.AggregatedMDMSolutions, error) {
+	if err := svc.authz.Authorize(ctx, &fleet.Host{TeamID: teamID}, fleet.ActionList); err != nil {
+		return nil, err
+	}
+
+	if teamID != nil {
+		_, err := svc.ds.Team(ctx, *teamID)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// it is expected that there will be relatively few MDM solutions. This
+	// returns the slice of all aggregated stats (one entry per mdm_id), and we
+	// then iterate to return only the one that was requested.
+	sols, _, err := svc.ds.AggregatedMDMSolutions(ctx, teamID)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, sol := range sols {
+		// don't take the address of the loop variable (although it could be ok
+		// here, but just bad practice)
+		sol := sol
+		if sol.ID == mdmID {
```

Comment:
Why aren't we doing this filtering at the ds level?

Reply:
We can't really do it easily - the mdm_id is part of the json blob stored in the row, only the type and id are available as columns. We could probably do some json querying in the select, but this is one area where support varies between mysql versions and possibly mariadb. Happy to reconsider if you have strong objections to this approach (I think `JSON_EXTRACT` would be the one to use in this case: https://dev.mysql.com/doc/refman/5.7/en/json-search-functions.html).

Reply:
It seems like it's supported in mariadb too: https://mariadb.com/kb/en/json_extract/

Reply:
Ah, didn't get that it's in json... in that case it doesn't sound as awkward, but maybe let's just add a comment?

Reply:
Updated with the comment.